sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -336,9 +336,11 @@ private[sql] object DataSourceScanExec {
   def create(
       output: Seq[Attribute],
       rdd: RDD[InternalRow],
-      relation: BaseRelation,
+      logicalRelation: LogicalRelation,
       metadata: Map[String, String] = Map.empty,
       metastoreTableIdentifier: Option[TableIdentifier] = None): DataSourceScanExec = {
+    val relation: BaseRelation = logicalRelation.relation
+
     val outputPartitioning = {
       val bucketSpec = relation match {
```

Comment:
What I mean is, we only enable bucketing(i.e. return `Some` here) if the conf is enabled, and the relation is bucketed, and the `output` contains all bucketing columns.  How about this?